pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -186,3 +243,39 @@ func (c *openServiceBrokerClient) DeleteServiceBinding(sID, bID string) error {
 
 	return nil
 }
+
+func (c *openServiceBrokerClient) pollBroker(ID string, operation string) error {
+	pollReq := brokerapi.LastOperationRequest{}
+	if operation != "" {
+		pollReq.Operation = operation
+	}
+
+	url := fmt.Sprintf(pollingFormatString, c.broker.Spec.URL, ID)
+	for i := 0; i < pollingAmountLimit; i++ {
+		glog.Infof("Polling attempt #%v: %s", i+1, url)
+		pollResp, err := util.SendRequest(c.client, http.MethodGet, url, pollReq)
+		if err != nil {
+			return err
+		}
+
+		lo := brokerapi.LastOperationResponse{}
+		if err := util.ResponseBodyToObject(pollResp, &lo); err != nil {
+			return err
+		}
+
+		switch lo.State {
+		case brokerapi.StateInProgress:
+		case brokerapi.StateSucceeded:
+			return nil
+		case brokerapi.StateFailed:
+			return errFailedState
+		default:
+			return errUnknownState
+		}
+
+		pollResp.Body.Close()
```

Comment:
Yeah, you're right. I wanted to close them once their utility is up, but missed that this loop will leave the last response unclosed. Will fix.